From 7da0d24619928690f4999ffc1693b643b612f130 Mon Sep 17 00:00:00 2001
From: Tomas Petr <tpetr@redhat.com>
Date: Thu, 11 May 2017 11:16:33 +0200
Subject: [PATCH] [ceph] exclude temporary mount locations from collection

Do not collect temporary mount paths at /var/lib/ceph/tmp/mnt.XXXX.

Resolves #1006
Replaces #1005

Signed-off-by: Tomas Petr <tpetr@redhat.com>
Signed-off-by: Bryn M. Reeves <bmr@redhat.com>
---
 sos/plugins/ceph.py | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/sos/plugins/ceph.py b/sos/plugins/ceph.py
index 1be7fd9..0b375d1 100644
--- a/sos/plugins/ceph.py
+++ b/sos/plugins/ceph.py
@@ -81,6 +81,13 @@ class Ceph(Plugin, RedHatPlugin, UbuntuPlugin):
         self.add_forbidden_path("/var/lib/ceph/*/*/*keyring*")
         self.add_forbidden_path("/var/lib/ceph/osd/*")
         self.add_forbidden_path("/var/lib/ceph/mon/*")
+
+# Excludes temporary ceph-osd mount location like
+# /var/lib/ceph/tmp/mnt.XXXX from sos collection.
+# In the /var/lib/ceph/tmp/ can still other files of potential
+# interest exists, so exclude only known temporary mount locations.
+
+        self.add_forbidden_path("/var/lib/ceph/tmp/*mnt*")
         self.add_forbidden_path("/etc/ceph/*bindpass*")
 
 # vim: set et ts=4 sw=4 :
-- 
2.7.4

